tests/shouldwork/Issues/T1921.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DerivingVia
+           , DerivingStrategies
+           , LambdaCase
+           , AllowAmbiguousTypes
+           , ApplicativeDo
+           , StandaloneDeriving
+           , StandaloneKindSignatures #-}
+
+module T1921 where
+
+import Clash.Prelude
+import Control.Lens
+import Data.Default
+import Data.Singletons.Prelude
+import Data.Singletons.TypeLits as TL
+
+topEntity :: Clock System -> Reset System -> Enable System -> Signal System (Unsigned 8)
+topEntity = exposeClockResetEnable fibonacciLFSR8
+
+-- Straightforward newtype
+type LFSRState :: Nat -> Type
+newtype LFSRState n = LFSRState { runLFSRState :: BitVector n }
+  deriving (NFDataX, AutoReg) via (BitVector n)
```

Comment:
I think we should remove `DerivingVia` and backport this to 1.4 in case there's another 1.4.x release

Reply:
Yeah, I removed it as part of getting it through CI